include/mitsuba/render/kdtree.h

```diff
@@ -542,6 +535,125 @@ template <typename BoundingBox_, typename Index_, typename CostModel_,
     /*                    Build-related data structures                     */
     /* ==================================================================== */
 
+    /* Append-only concurrent vector, whose storage is arranged into slices
```

Comment:
A small request: A lot of this code technicaly doesn't depend on any KDTree template parameters. Can you move it to a `detail::` namespace? That includes `OrderedChunkAllocator` and `concurrent_vector`, maybe others as well. For `concurrent_vector`, it should probably renamed `ConcurrentVector` for consistency.

Reply:
`Index` and `Size` should remain `uint32_t` types here, because this code really will not work anymore if you change to `uint64_t`